common/Mesh-Network-Joining-Process-Comparison.md

```diff
@@ -0,0 +1,540 @@
+An FFD is a device that is capable of serving as a personal area network (PAN) coordinator or a coordinator.   
+An RFD is a device that is not capable of serving as either a PAN Coordinator or a coordinator. An RFD is intended for applications that are extremely simple, such as a light switch or a passive infrared sensor; it does not have the need to send large amounts of data and only associates with a single FFD at a time. Consequently, the RFD can be implemented using minimal resources and memory capacity.   
+
+### 2.2.2. Star and Peer-to-Peer topology
+Depending on the application requirements, an IEEE 802.15.4 low-rate wireless personal area network (LR-WPAN) operates in either star topology or peer-to-peer topology as shown in the figure below.   
+<div align="center">
+  <img src="files/CM-Mesh-Network-Comparison/joining-process-802154-topology.png">  
+</div>  
+
+In the star topology, the communication is established between devices and a single central controller, called the Personal Area Network (PAN) Coordinator. The PAN Coordinator is the primary controller of the PAN.   
+
+The peer-to-peer topology also has a PAN Coordinator, however, it differs from the star topology in that any device is able to communicate with any other device as long as they are in range of one another. Peer-to-peer topology allows more complex network formations to be implemented, such as mesh networking topology. 
+A peer-to-peer network allows multiple hops to route messages from any device to any other device on the network. Such functions can be added at the higher layer, which is Network layer in Zigbee architecture.   
+
+The PAN Coordinator forms a network by choosing an unused PAN ID and broadcasting beacon frames to neighboring devices. Once the PAN ID is chosen, the PAN Coordinator allows other devices, potentially both FFDs and RFDs, to join its network. A candidate device receiving a beacon frame is able to request to join the network at the PAN Coordinator. If the PAN Coordinator permits the device to join, it adds the new device as a child device in its neighbor list. Then the newly joined device adds the PAN Coordinator as its parent in its neighbor list and begins transmitting periodic beacons; other candidate devices are able to then join the network at that device. If the original candidate device is not able to join the network at the PAN Coordinator, it will search for another parent device.   
+
+
+## 2.3. Mesh Network maintenance
+The network formation is performed by next higher layer which is network layer, and we will discuss how to maintain the network from network layer view.   
+
+### 2.3.1. Device type in Zigbee network
+
+* **PAN coordinator**: the principal controller of an IEEE 802.15.4-based network that is responsible for network formation. The PAN coordinator must be a full function device (FFD). 
+* **ZigBee coordinator**: an IEEE 802.15.4 PAN coordinator. 
```

Comment:
1. [Mark] What is the difference between PAN coordinator and Zigbee Coordinator. 
2. ZigBee 
[Mark] Is the Zigbee a better choice?

Reply:
Actually, Zigbee coordinator is the PAN coordinator in IEEE 802.15.4. For avoiding confusion, I will only keep Zigbee coordinator here, and move the description of PAN coordinator under Zigbee coordinator.